test/libsolidity/util/TestFunctionCall.h

```diff
@@ -95,6 +97,8 @@ class TestFunctionCall
 	void setContractABI(Json::Value _contractABI) { m_contractABI = std::move(_contractABI); }
 	void setSideEffects(std::vector<std::string> _sideEffects) { m_call.actualSideEffects = _sideEffects; }
 
+	bool valid() const { return matchesExpectation() && validateFractionDigits(); }
```

Comment:
I'm not sure if `valid` conveys that it also has to match the expectation.

Also does this need to be public? `matchesExpectation` is private.

Reply:
Yes it need to be public, because it need to get called in `SemanticTest::runTest` to define `success`. 

However, I did now a minor refactoring: `valid()` is now private and only used by the formatting function `TestFunctionCall::format(..)`. Where I changed the visibility of `validFractionDigits` to be public, so this function can get called by `SemanticTest::runTest`. The advantage is that we now only check the fraction digits here, no need to check whether the expectation is matching here.

That means if the result uses fixed point types, where the fraction digits does not match, the test-case will fail.

Reply:
Ok I just removed that now.